### ISSUE TITLE
use in test not finding crate 

### ISSUE THREAD

**Issue by jerusdp** [rust-lang/rust#89552]
I tried this code using Rust beta (Edition 2021):

```rust
use hcaptcha::Hcaptcha;
// use hcaptcha_derive::Hcaptcha;

#[derive(Hcaptcha)]
enum Test {
    #[captcha]
    Hcaptcha(String),
}

fn main() {
    println!("Super!");
}
```

Which is run by a test run in the `tests` folder: 

```rust
#[test]
fn ui() {
    let t = trybuild::TestCases::new();
    t.compile_fail("tests/ui/*.rs");
}

```

I expected to see this happen: 

I expected an error message from the derive macro as follows:

```rust
error: Must derive on a struct

  = help: This macro can only be implemented on a struct.

        #[derive(Hcaptcha)]
        struct MyStruct {
            #[captcha]
            hcaptcha: String,
        }

 --> $DIR/not_a_struct.rs:5:6
  |
5 | enum Test {
  |      ^^^^

```
This message is the one generated when the code is compiled as Edition2018 (also Beta channel).

Instead, this happened:

The 'hcaptcha' crate was not found as a dependency.

```rust
error[E0432]: unresolved import `hcaptcha`
 --> $DIR/not_a_struct.rs:1:5
  |
1 | use hcaptcha::Hcaptcha;
  |     ^^^^^^^^ use of undeclared crate or module `hcaptcha`

error: cannot determine resolution for the derive macro `Hcaptcha`
 --> $DIR/not_a_struct.rs:4:10
  |
4 | #[derive(Hcaptcha)]
  |          ^^^^^^^^
  |
  = note: import resolution is stuck, try simplifying macro imports

error: cannot find attribute `captcha` in this scope
 --> $DIR/not_a_struct.rs:6:7
  |
6 |     #[captcha]
  |       ^^^^^^^
```
I get the same result from the current nighlty build. 

@dtolnay 's trybuild may be a contributor to this behaviour. The testing process is patterned on that used by the serde project. 

### Meta

`rustc --version --verbose`:
```
rustc 1.56.0-beta.3 (deef86610 2021-09-17)
binary: rustc
commit-hash: deef866109022248b2506587b377c6b1c881d963
commit-date: 2021-09-17
host: x86_64-unknown-linux-gnu
release: 1.56.0-beta.3
LLVM version: 13.0.0
```

<details><summary>Backtrace</summary>
<p>

No backtrace generated.

```
<backtrace>
```

</p>
</details>

**Comment by ehuss** [comment]
Thanks for the report!  Can you maybe provide a fully working example that I could try locally?  I don't see `hcaptcha_derive` existing anywhere, so maybe this is something you are working on?

I suspect the issue from the error message is that this is maybe due to a change in the resolver.  Features aren't unified with build dependencies, and maybe the existence of `hcaptcha::Hcaptcha` is gated by a feature that is not enabled?

**Comment by jerusdp** [comment]
@ehuss You are quite right, it is a work in progress. 

I have pushed the code to the "traits" branch here: https://github.com/jerusdp/hcaptcha-rs/tree/traits
(Currently with Edition2018 as I am developing tests).

Neither the trait nor the derive macro are gated behind a feature (yet, and maybe never 😃 ).

**Comment by ehuss** [comment]
Can you include exactly which commands you ran to get the unresolved import? And did you just change the edition on all the packages to 2021? Or are there other changes needed?

**Comment by ehuss** [comment]
Hm, I'm unable to reproduce with that with commit e3f3a5bd0d7b09929733b4d8e51c92475e36a404.  Setting the edition of every package to 2021, and running that command I get:

* 1.56.0-beta.3 or 1.56.0-beta.4: tests pass
* rustc 1.57.0-nightly (25ec82738 2021-10-05): I get this error:

```
test tests/ui/not_a_struct.rs ... mismatch

EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: Must derive on a struct

  = help: This macro can only be implemented on a struct.

        #[derive(Hcaptcha)]
        struct MyStruct {
            #[captcha]
            hcaptcha: String,
        }

 --> $DIR/not_a_struct.rs:5:6
  |
5 | enum Test {
  |      ^^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error: Must derive on a struct
 --> $DIR/not_a_struct.rs:5:6
  |
5 | enum Test {
  |      ^^^^
  |
  = help: This macro can only be implemented on a struct.

                  #[derive(Hcaptcha)]
                  struct MyStruct {
                      #[captcha]
                      hcaptcha: String,
                  }
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
note: If the actual output is the correct output you can bless it by rerunning
      your test with the environment variable TRYBUILD=overwrite



test ui ... FAILED
```

I notice that the `Cargo.lock` is not checked in, so perhaps you are using older versions of dependencies that are somehow causing a problem?  Does a `cargo update` help?
